tests/testcases/key-value-undefined-store/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+use anyhow::{ensure, Result};
+use itertools::sorted;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+    key_value::{Error, Store},
+};
+
+#[http_component]
+fn handle_request(req: Request) -> Result<Response> {
+    // TODO: once we allow users to pass non-default stores, test that opening
+    // an allowed-but-non-existent one returns Error::NoSuchStore
+    ensure!(matches!(Store::open("forbidden"), Err(Error::AccessDenied)));
+
+    let query = req.uri().query().expect("Should have a testkey query string");
+    let query: std::collections::HashMap::<String, String> = serde_qs::from_str(query)?;
+    let init_key = query.get("testkey").expect("Should have a testkey query string");
+    let init_val = query.get("testval").expect("Should have a testval query string");
+
+    let store = Store::open_default()?;
+
+    store.delete("bar")?;
+
+    ensure!(!store.exists("bar")?);
+
+    ensure!(matches!(store.get("bar"), Err(Error::NoSuchKey)));
```

Comment:
I personally find it easier to read long tests like this if assertions are grouped with the side-effect-inducing call they are testing, e.g.:

```suggestion
    store.delete("bar")?;
    ensure!(!store.exists("bar")?);
    ensure!(matches!(store.get("bar"), Err(Error::NoSuchKey)));
```

Reply:
actually none of this test body is needed because `spin up` will barf during validation and never actually run it so I just deleted it all.  Wake the heck up, Ivan